### PR TITLE
Refactoring convert code so it can be used outside utils package

### PR DESCRIPTION
**Pull request by girishramnani**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

/kind code-refactoring

**What does does this PR do / why we need it**:
Refactoring convert code so it can be used outside utils package. This helps towards https://github.com/openshift/odo/issues/4281

**Which issue(s) this PR fixes**:

partly https://github.com/openshift/odo/issues/4281

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [X] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
everything should work the same

### CODE REVIEW

**Comment by girishramnani** [comment]
this would allow us to write unit test for convert.go code as well

File: pkg/devfile/convert/convert.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+}
+
+// setDevfileComponentsForS2I sets the devfile.yaml components field from s2i data.
+func setDevfileComponentsForS2I(d data.DevfileData, s2iImage string, localConfig *config.LocalConfigInfo, s2iEnv config.EnvVarList) error {
+	klog.V(2).Info("Set devfile components from s2i data")
+
+	maxMemory := localConfig.GetMaxMemory()
+	volumes, _ := localConfig.ListStorage()
+	urls, _ := localConfig.ListURLs()
```
Comment (mik-dass) [inline pkg/devfile/convert/convert.go:204]: Why are we ignoring the errors here?

**Comment by girishramnani** [comment]
I am just getting timeouts on this.

**Review by mik-dass — APPROVED** [review]
/lgtm

**Pull request merged**